Revisit the RH strings occurences

hustodemon opened the pull request:
## GUI diff

No difference.


- [x] **DONE**

## Documentation

- [ ] **DONE**

## Test coverage
- No tests:  nothing to test here

- [x] **DONE**

## Links

https://github.com/SUSE/spacewalk/issues/12289

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)


## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

mcalmer commented on client/rhel/spacewalk-client-tools/src/up2date_client/up2dateErrors.py:
```diff
@@ -190,6 +190,7 @@ def __repr__(self):
     def changeExplanation(self):
+        # TODO no idea
         return _("""
 Red Hat Network Classic is not supported.
```
Maybe we shorten it and say

   Red Hat Network Classic is not supported in Uyuni.

And remove the rest. Whatever "Red Hat Network Classic" is, but we say this method is not supported :-)

hustodemon replied:
Alright :smile:

mcalmer commented on java/code/src/com/redhat/rhn/frontend/strings/java/StringResource_en_US.xml:
```diff
@@ -1921,19 +1921,19 @@
 &lt;ul&gt;
 &lt;li&gt;
-Is the 'rhn-tools' child software channel for this kickstart profile's base
+Is the tools child software channel for this kickstart profile's base
```
It sound not good.

Should we just call it "Client Tools Channel" and remove the "software"? Also in the text below?

hustodemon replied:
That sounds good.

mcalmer commented on java/code/src/com/redhat/rhn/frontend/strings/java/StringResource_en_US.xml:
```diff
@@ -8648,7 +8648,7 @@ Alternatively, you will want to download &lt;strong&gt;Incremental Channel Conte
         <source>No module filters attached. Module metadata will be cloned to target.</source>
         <context-group name="ctx">
-          <context context-type="sourcefile">com.redhat.rhn.domain.contentmgmt.validation.ModularSourcesValidator</context>
+          <context context-type="sourcefile">com.redhat.hn.domain.contentmgmt.validation.ModularSourcesValidator</context>
```
this seems to be wrong

hustodemon replied:
Hopalla. I made a mistake during conflict resolution. Thanks!

mcalmer commented on java/code/src/com/redhat/rhn/frontend/strings/jsp/StringResource_en_US.xml:
```diff
@@ -6314,8 +6314,8 @@ organization. You may grant or remove the organization administrator role in the
-                However, it is important to keep in mind that requesting a 
-                lock might be delayed. If a package has been updated 
+                However, it is important to keep in mind that requesting a
+                lock might be delayed. If a package has been updated
```
I hope the trailing whitespaces are really not needed. If possible check the text in the UI

hustodemon replied:
I reverted this change. It's not necessary.

hustodemon commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -19,77 +19,76 @@
-CONNECT_WINDOW_TEXT   = _("We are attempting to contact the Red Hat "
-                          "Network server at %s.")
+CONNECT_WINDOW_TEXT   = _("We are attempting to contact the Uyuni "
+                          "server at %s.")
```
This message is used in the TUI interface of the `up2date_client`: https://github.com/uyuni-project/uyuni/blob/master/client/rhel/spacewalk-client-tools/src/up2date_client/tui.py#L334

To me it looks like Uyuni/Spacewalk/Satellite/SUSE Manager is contacted here, not the RHN/SCC, so I think the change is correct.

mcalmer replied:
The problem is, RHN is a hosted version of Satellite. So it is hard to know if this apply also to Uyuni, or they request a task which can be done only in SCC when we talk about SUSE.
Ok with me.

hustodemon commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -19,77 +19,76 @@
 
 # Why Register Window
-WHY_REGISTER          = _("Why Should I Connect to RHN? ...")
+WHY_REGISTER          = _("Why Should I Connect to Uyuni? ...")
```
Again, this is in used in `rhn_register` and I think Uyuni is the correct entity here. When you click the button `Why should I connect to RHN`, you get a list of advantages of registering to satellite.

See the screenshot:

![ss-rhn](https://user-images.githubusercontent.com/1412268/93879262-b4d33780-fcdb-11ea-9939-fa4baaf03925.png)

after clicking:
![ss-rhn2](https://user-images.githubusercontent.com/1412268/93879361-dd5b3180-fcdb-11ea-9484-c0db8cccb6b0.png)

mcalmer replied:
ok

paususe replied:
@hustodemon 

Do we still use `rhn_register` at all?

hustodemon replied:
I have no idea, but to determine if we use some codepath is a different task.

hustodemon commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -19,77 +19,76 @@
-LOGIN_PROMPT      = _("Please enter your login information for the %s Red "
-                    "Hat Network Satellite:\n\n")
-HOSTED_LOGIN      = _("Red Hat Login:")
+REGISTER_WINDOW   = _("Uyuni Account")
+LOGIN_PROMPT      = _("Please enter your login information for the %s Uyuni:\n\n")
+HOSTED_LOGIN      = _("Uyuni Login:")
```
No, these are SUMA/Uyuni credentials. See here:

![ss-rhn3](https://user-images.githubusercontent.com/1412268/93879506-1a272880-fcdc-11ea-863e-d313dbfa00c3.png)

mcalmer replied:
REGISTER_WINDOW and LOGIN_PROMPT seems to be correct.
But HOSTED_LOGIN is not in this screenshot.

hustodemon commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -134,53 +133,52 @@
-                            "via the Red Hat Satellite web interface.  (Instructions for this are in chapter 6 "
-                            "of the RHN Reference Guide, available from the 'Help' button in the main Red "
-                            "Hat Network Satellite web interface.)") % PM_NAME
+                            "via the Uyuni web interface.  (Instructions for this are in chapter 6 "
```
Agreed. I generalized the text a bit, so that it still makes sense and is useful.

hustodemon commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -193,17 +191,17 @@
+"Contact your Uyuni administrator about this problem. Once you make the "
 "appropriate active subscriptions available in your account, you may browse "
-"to this system's profile in the RHN web interface and subscribe this system "
+"to this system's profile in the Uyuni web interface and subscribe this system "
```
I think Uyuni/SUMA is correct here: if you read the whole message, I believe "they" are making reference to Satellite UI:

> "Contact your Uyuni administrator about this problem. Once you make the "
"appropriate active subscriptions available in your account, you may browse "
"to this system's profile in the Uyuni web interface and subscribe this system "
"to **software channels via the software > software channels** tab.")

mcalmer replied:
ok

hustodemon commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -193,17 +191,17 @@
 SLOTS                   =  SLOTS_TITLE + "\n" + OK_SLOTS + "\n%s"
 FAILED_SLOTS            = _("This system was unable to be associated with the "
-                            "following RHN module(s) because there were "
+                            "following Uyuni module(s) because there were "
```
Yes, this sounds a bit fishy. I'd keep RHN modules then.

hustodemon replied:
Hm, after studying the source, it seems that slots are entitlements and this message pops up when user runs out of them (shouldn't happen in SUMA/Uyuni).

hustodemon commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -215,40 +213,40 @@
 NO_SYS_ENTITLEMENT         = _("This system was unable to be associated with "
-"any RHN service level modules. This system will not receive any software "
-"updates to keep it secure and supported. Contace your Satellite administrator "
+"any Uyuni service level modules. This system will not receive any software "
```
Seems it's the same as above.

hustodemon commented:
Thanks for the comments @mcalmer . I've gone throught them and addressed some. For the rest I added an explanation (and screenshots). Please take a look at them and lemme know, what you think.

mcalmer reviewed (approved):
Most of them should be ok now. Thanks

mcalmer commented:
@hustodemon don't forget to port to 4.1

mcalmer commented:
(maybe even to 4.0?)

paususe commented on client/rhel/spacewalk-client-tools/src/firstboot-legacy-rhel5/rhn_login_gui.py:
```diff
@@ -40,9 +40,9 @@
+    moduleName = _("Login")
     windowTitle = moduleName
-    shortMessage = _("Register with Red Hat Satellite")
+    shortMessage = _("Register with Uyuni")
```
Is this string even used these days? I'm surprised this string is in the Python code but not in https://github.com/uyuni-project/uyuni/blob/master/client/rhel/spacewalk-client-tools/po/rhn-client-tools.pot

Maybe our string-extraction is not looking at this "firstboot-legay-rhel5" directory? Should it? According to the [Makefile](https://github.com/uyuni-project/uyuni/blob/master/client/rhel/spacewalk-client-tools/src/firstboot-legacy-rhel5/Makefile) in that directory, that installs "man pages for the up2date client". Do we use the up2date  client at all?

hustodemon replied:
> Is this string even used these days? 
I don't know. I didn't analyze every one string usage here, the task would take even longer.

> Do we use the up2date client at all?
AFAIK yes. up2date client _is_ the traditional stack.

paususe commented on client/rhel/spacewalk-client-tools/src/firstboot-legacy-rhel5/rhn_provide_certificate_gui.py:
```diff
@@ -34,7 +34,7 @@ class RhnProvideCertificateWindow(RhnRegisterFirstbootGuiWindow, rhnregGui.Provi
     moduleName = _("Provide Certificate")
     windowTitle = moduleName
-    shortMessage = _("Provide a certificate for this Red Hat Satellite server")
+    shortMessage = _("Provide a certificate for this Uyuni server")
```
Same observation as above. In this case, it's easy to suggest something downstream-neutral.

```suggestion
    shortMessage = _("Provide a certificate for this Server")
```

paususe commented on client/rhel/spacewalk-client-tools/src/firstboot-legacy-rhel5/rhn_review_gui.py:
```diff
@@ -32,7 +32,7 @@ class RhnReviewWindow(RhnRegisterFirstbootGuiWindow, rhnregGui.ReviewSubscriptio
     moduleName = _("Review Subscription")
     windowTitle = moduleName
-    shortMessage = _("Connect to Red Hat Satellite")
+    shortMessage = _("Connect to Uyuni")
```
```suggestion
    shortMessage = _("Connect to the Server")
```

And anyway, same comment as above: I think this code is not used :-?

hustodemon replied:
Again: no idea.

paususe commented on client/rhel/spacewalk-client-tools/src/firstboot-legacy-rhel5/rhn_start_gui.py:
```diff
@@ -33,7 +33,7 @@ class RhnStartWindow(RhnRegisterFirstbootGuiWindow):
     moduleName = _("Set Up Software Updates")
     windowTitle = moduleName
-    shortMessage = _("Register with Red Hat Satellite")
+    shortMessage = _("Register with Uyuni")
```
Same comment

paususe commented on client/rhel/spacewalk-client-tools/src/firstboot-legacy-rhel6/rhn_login_gui.py:
```diff
@@ -35,8 +35,8 @@ class moduleClass(Module):
         self.priority = 108.5
-        self.sidebarTitle = _("Red Hat Account")
-        self.title = _("Red Hat Account")
+        self.sidebarTitle = _("Uyuni Account")
+        self.title = _("Uyuni Account")
```
Same. It seems this was for users logging in to RHN. That would be SCC in SUSE, or nothing in Uyuni. But probably we don't use this codepath anymore :-?

hustodemon replied:
I don't think that the client allowed direct connection to RHN and I think that this is the same as the other cases above, when this really meant Satellite, not RHN.

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -19,77 +19,76 @@
 # Satellite URL Window
-SATELLITE_URL_WINDOW  = _("Enter your Red Hat Satellite URL.")
-SATELLITE_URL_TEXT    = _("Please enter the location of your Red Hat Satellite "
+SATELLITE_URL_WINDOW  = _("Enter your Uyuni URL.")
+SATELLITE_URL_TEXT    = _("Please enter the location of your Uyuni "
```
```suggestion
SATELLITE_URL_WINDOW  = _("Enter your Server URL.")
SATELLITE_URL_TEXT    = _("Please enter the location of your Server "
```

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -19,77 +19,76 @@
                           "certificate. The SSL certificate is only required "
                           "if you will be connecting over https (recommended).")
-SATELLITE_URL_PROMPT  = _("Satellite URL:")
+SATELLITE_URL_PROMPT  = _("Uyuni URL:")
```
```suggestion
SATELLITE_URL_PROMPT  = _("Server URL:")
```

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -19,77 +19,76 @@
+SATELLITE_URL_PROMPT  = _("Uyuni URL:")
 SATELLITE_URL_PROMPT2 = _("SSL certificate:")
-SATELLITE_REQUIRED    = _("You must enter a valid Satellite URL.")
+SATELLITE_REQUIRED    = _("You must enter a valid Uyuni URL.")
```
```suggestion
SATELLITE_REQUIRED    = _("You must enter a valid Server URL.")
```

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -19,77 +19,76 @@
 CONNECT_WINDOW        = _("Attempting to contact the Spacewalk server.")
-CONNECT_WINDOW_TEXT   = _("We are attempting to contact the Red Hat "
-                          "Network server at %s.")
+CONNECT_WINDOW_TEXT   = _("We are attempting to contact the Uyuni "
+                          "server at %s.")
```
```suggestion
CONNECT_WINDOW_TEXT   = _("We are attempting to contact the Server "
                          "at %s.")
```

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -19,77 +19,76 @@
 START_REGISTER_TEXT   = _("This assistant will guide you through "
                           "connecting your system to "
-                          "Red Hat Satellite to receive software "
+                          "Uyuni to receive software "
```
```suggestion
                          "Uyuni to receive software "
```

Definitely need a PRODUCT_NAME replacement to make this unsurprising for downstream users :-(

hustodemon replied:
Done.

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -19,77 +19,76 @@
 PASSWORD          = _("Password:")
 LOGIN_TIP         = _("Tip: Forgot your login or password?  Contact your "
-                      "Satellite's Organization Administrator.")
+                      "Uyuni Organization Administrator.")
```
```suggestion
                      "Organization Administrator.")
```

Downstream users may not know who the "Uyuni Organization Administrator" is. They will know about the "SUSE Manager Organization Administrator".

"Organization Administrator" is not good but IMHO it's less bad than "Uyuni Organization Administrator".

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -193,62 +191,62 @@
-"any RHN service level modules. This system will not receive any software "
-"updates to keep it secure and supported. Contace your Satellite administrator "
+"any Uyuni system types. This system will not receive any software "
+"updates to keep it secure and supported. Contace your Uyuni administrator "
```
```suggestion
"updates to keep it secure and supported. Contact your Uyuni administrator "
```

I skipped a lot of the above changes but same comments as before:
a) I wonder if we really run this code these days
b) If we do, we should have a PRODUCT_NAME replacement with either "Uyuni" or "SUSE Manager"

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -193,62 +191,62 @@
                             "installation number.  "
-                            "You may also refer to the RHN Reference Guide, section 6.4.6 for more details "
-                            "about activation keys (http://access.redhat.com/knowledge/docs/Red_Hat_Network/)\n"
+                            "You may also refer to the Uyuni Reference Guide for more details "
+                            "about activation keys (https://www.uyuni-project.org/uyuni-docs/uyuni/reference/intro.html)\n"
```
This will definitely look weird to downstream users, referring them to Uyuni docs instead of SUSE Manager docs :-/

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -260,7 +258,7 @@
 SSL_CERT_ERROR_MSG         = _("<b><span size=\"16000\">Incompatible Certificate File</span></b>\n\n"
                                "The certificate you provided, <b>%s</b>, is not compatible with "
-                               " the Red Hat Satellite server at <b>%s</b>. You may want to double-check"
+                               " the Uyuni server at <b>%s</b>. You may want to double-check"
```
```suggestion
                               " the Server at <b>%s</b>. You may want to double-check"
```

Since we do not have PRODUCT_NAME, please try to use something as neutral as possible (i. e. avoid "Uyuni" or "SUSE Manager"). There are some more opportunities above where I did not comment.

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/rhnreg_constants.py:
```diff
@@ -273,21 +271,18 @@
                                   "/etc/sysconfig/rhn/up2date. You can either make the "
                                   "serverURL use http instead of https, or you can "
-                                  "download the SSL cert from your Satellite, place it "
+                                  "download the SSL cert from your Uyuni, place it "
```
```suggestion
                                  "download the SSL cert from your Server, place it "
```

paususe commented on client/rhel/spacewalk-client-tools/src/up2date_client/up2dateErrors.py:
```diff
@@ -191,12 +191,7 @@ def __repr__(self):
-    subscription-manager register --auto-attach
-
-Get more information at access.redhat.com/knowledge
+Red Hat Network Classic is not supported in Uyuni.
```
```suggestion
Red Hat Network Classic is not supported in Uyuni or SUSE Manager.
```

hustodemon commented on backend/satellite_tools/satsync.py:
```diff
@@ -532,8 +532,8 @@ def initialize(self):
                 log(-1, _("ERROR: Live content synchronizing with RHN Classic Hosted is no longer supported.\nPlease "
-                          "use the cdn-sync command instead unless you are attempting to sync from another Satellite "
-                          "via Inter-Satelite-Sync (ISS), or from local content on disk via Channel Dump ISOs."),
+                          "use the cdn-sync command instead unless you are attempting to sync from another Uyuni "
```
I know that we have a product name placeholder in **java**, but not in python. Could you point me to one?

hustodemon replied:
Yes - your PR only has the placeholders for java, afaik. But not for python.

hustodemon commented:
Implemented the `PRODUCT_NAME` placeholder for python code. @brejoc or @agraul , could you take a look at it?

brejoc reviewed (approved):
lgtm!

paususe reviewed (approved):
Huge and great effort, thank you!

hustodemon commented:
Thanks everybody for your reviews! Merging & backporting now.

paususe commented:
@hustodemon 
Has this been merged in Uyuni master? It seems not :-?
